test/run_test.py

```diff
@@ -0,0 +1,253 @@
+    else:
+        print ("Error: problem occurred while looking for dnf or yum")
+        sys.exit(1)
+
+    # add all basic packages
+    for pck in needed_packages:
+        cmd_download = cmd_download + " "+pck
+
+    # download packages
+    child = Popen(cmd_download, shell=True, stdout=PIPE, stderr=PIPE)
+    stdout, stderr = child.communicate()
+    if child.returncode:
+        print ("ERROR: during downloading packages this errors occurred:")
+        print (stderr)
+
+    # create repo from downloaded packages
+    createrepo_cmd = ["createrepo", repo_path]
+    child = Popen(createrepo_cmd, stdout=PIPE, stderr=PIPE)
+    stdout, stderr = child.communicate()
+    if stderr:
+        print ("ERROR: during creating repo this errors occurred:")
+        print (stderr)
+
+
+def delete_repo(releaseid):
+    """Remove repo.
+
+    Note: deletes repo even though, if was not created by this script
+    """
+    print ("Deleting repo")
+    conf = config.Config()
+    repo_path = os.path.join(conf["RepoDir"], releaseid)
+    if not os.path.isdir(repo_path):
+        return
+    else:
+        shutil.rmtree(repo_path)
+
+
+def generate_coredump():
+    """Generate coredump.
+
+    This is done by calling gcore on this script.
+    """
+    print ("Creating coredump")
+    # use gcore on this process
+    gcore_cmd = ['gcore', '-o', "/var/tmp/coredump", str(os.getpid())]
+    child = Popen(gcore_cmd, stdout=PIPE, stderr=PIPE)
+    stdout, stderr = child.communicate()
+    if child.returncode:
+        print ("ERROR: following error was generated by gcore")
+        print (stderr)
+
+    # get only last line form output - there is filename mentioned
+    lastline = stdout.splitlines()[-1]
+    # get the last word, that should be coredump path
+    corepath = lastline[lastline.rfind(" ")+1:]
+    if not corepath.startswith("/var/tmp"):
+        print ("ERROR: corefile was not generated")
+        sys.exit(1)
+
+    # rename it, so it is called coredump, not coredump.XXXXX
+    os.rename(corepath, corepath[:corepath.find(".")])
+    corepath = corepath[:corepath.find(".")]
+
+    # find in which package is python
+    exe_link = os.readlink('/proc/self/exe')
+    rpm_cmdline = 'rpm -qf '+exe_link
+    child = Popen(rpm_cmdline, shell=True, stdout=PIPE, stderr=PIPE)
+    package, stderr = child.communicate()
+    if child.returncode:
+        print ("ERROR: rpm for python was not found.")
+        sys.exit(1)
+    package = package[:package.rfind(".")]
+    return (corepath, package, sys.executable)
+
+
+def create_local_config(change_repo_dir):
+    """Create local config file, change what is needed and edit env var."""
+    # create local copy of set configuration file
+    # read current configuration file
+    config_path = os.environ.get('RETRACE_SERVER_CONFIG_PATH')
+    new_conf_path = '/var/tmp/retrace-server.conf'
+    with open(config_path) as f:
+        lines = f.read().splitlines()
```

Comment:
there is difference between my approach and yours. Your way keeps trailing \n, mine don't.